src/Bicep.Core.IntegrationTests/LambdaTests.cs

```diff
@@ -137,8 +137,8 @@ public void Map_function_works_with_unknowable_types()
             var info = result.GetInfoAtCursors(cursors);
 
             info.Should().SatisfyRespectively(
-                x => x.Type.Name.Should().Be("any[]"),
-                x => x.Type.Name.Should().Be("any"),
+                x => x.Type.Name.Should().Be("(never)[]"),
```

Comment:
Possible to avoid the parentheses when generating the type name for an array of `never`? IIRC we have code to differentiate in other cases.